src/os_posix/os_open.c

```diff
@@ -30,180 +84,77 @@ __open_directory(WT_SESSION_IMPL *session, char *path, int *fd)
  */
 int
 __wt_open(WT_SESSION_IMPL *session,
-    const char *name, bool ok_create, bool exclusive, int dio_type, WT_FH **fhp)
+    const char *name, int dio_type, u_int flags, WT_FH **fhp)
```

Comment:
I like that you got rid of multiple bool args and replaced it with flags.  I would've made it a uint32_t myself.